crates/turbo-tasks-memory/src/scope.rs

```diff
@@ -681,12 +645,18 @@ impl TaskScopeState {
         match self.collectibles.entry(trait_id) {
             Entry::Occupied(mut entry) => {
                 let (collectibles, dependent_tasks) = entry.get_mut();
-                if collectibles.remove_count(collectible, count) {
+                let old_value = collectibles.get(&collectible);
+                let new_value = old_value - count as isize;
+                if collectibles.remove_count(collectible, count) || new_value < 0 {
```

Comment:
Why the check for `new_value < 0` here? Intuitively it seems like `remove_count` should take care of it.

Reply:
Yeah that was a bug. The read_collectibles need to be invalidated when negative count changes. Each negative count will eliminate one child scope emitted collectible. So changing from -1 to -2 might affect the visible collectibles.

Reply:
Can you add a comment to explain this here?